pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/ReaderThread.java

```diff
@@ -118,6 +119,11 @@ public void run() {
     }
 
     protected void createActualReader() throws org.apache.pulsar.client.api.PulsarClientException, ExecutionException {
+        // As `authentication` is not serializable, it will be null after the client config is
+        // deserialized, which results in that the `getAuthentication` return a new instance of
+        // `AuthenticationDisabled`.
+        // To hit the guava cache, we should set the authentication to the static instance.
+        clientConf.setAuthentication(AuthenticationDisabled.INSTANCE);
```

Comment:
Doesn't this change completely disable authentication?

Reply:
It will be reconstructed by `authPluginClassName` and `authParams`. I have fixed the comments.